include/util/query_heap.hpp

```diff
@@ -193,7 +193,20 @@ template <typename NodeID,
 class QueryHeap
 {
   private:
-    using HeapData = std::pair<Weight, Key>;
+    struct HeapData
+    {
+        Weight weight;
+        Key index;
+
+        bool operator>(const HeapData &other) const
+        {
+            if (weight == other.weight)
```

Comment:
It seems it guarantees that we will get keys with the same weight in insertion order and I added it only to preserve guarantees we had before with std::pair, but is it really needed? 🤔